src/main/java/nextstep/subway/domain/LineStations.java

```diff
@@ -0,0 +1,108 @@
+package nextstep.subway.domain;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToMany;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Embeddable
+public class LineStations {
+    @OneToMany(cascade = {CascadeType.MERGE, CascadeType.PERSIST}, orphanRemoval = true)
+    @JoinColumn(name = "line_id")
+    private List<LineStation> lineStations = new ArrayList<>();
+
+    public void add(Station upStation, Station downStation, Long distance) {
+        if (this.lineStations.isEmpty())  {
+            lineStations.add(new LineStation(distance, upStation, downStation, true));
+            return;
+        }
+        LineStation addTarget = findAddTarget(upStation, downStation);
+
+        checkPossibleAddSection(addTarget, distance);
+
+        this.lineStations.add(
+                addTarget.addStation(upStation, downStation, distance)
+        );
+    }
+
+    public List<Station> getStationsSortedByUpToDown() {
+        List<Station> result = new ArrayList<>();
+        LineStation lineStation = findStartLineStation();
+
+        result.add(lineStation.getUpStation());
+        result.add(lineStation.getDownStation());
+
+        for (int i = 0; i < this.lineStations.size() - 1; i++) {
+            lineStation = findNextLineStation(lineStation.getDownStation());
+            result.add(lineStation.getDownStation());
+        }
+
+        return result;
```

Comment:
```suggestion
        return lineStations.stream()
            .map(it -> findNextLineStation(it.getDownStation()))
            .map(LineStation::getDownStation)
            .collect(Collectors.toList());
```

이렇게도 해볼수 있을것 같아요. 참고만해주세요~